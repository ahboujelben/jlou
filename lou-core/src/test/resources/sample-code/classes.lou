class HelloClass {
  init() {
    this.field = "test field";
  }
  printMe() {
    print "I'm a method in a class";
  }
}

print HelloClass;
var testInstance = HelloClass();
print testInstance.field;
testInstance.field = 2;
print testInstance.field;
print testInstance.printMe;
testInstance.printMe();

class Animal {
  run() {
    print "animal running";
  }
}

var tiger = Animal();
tiger.run();

class Cake {
  taste() {
    var adjective = "delicious";
    print "The " + this.flavor + " cake is " + adjective + "!";
  }
}

var cake = Cake();
cake.flavor = "German chocolate";
cake.taste();

class Circle {
  init(radius) {
    this.radius = radius;
  }

  getArea() {
    return 3.141592653 * this.radius * this.radius;
  }
}

var circle = Circle(4);
print circle.getArea();
