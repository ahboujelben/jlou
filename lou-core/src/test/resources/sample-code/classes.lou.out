<class HelloClass>
test field
2
<fn printMe>
I'm a method in a class
animal running
The German chocolate cake is delicious!
50.265482448